test/functional/test_framework/test_framework.py

```diff
@@ -477,32 +486,24 @@ def _initialize_chain(self):
             for node in self.nodes:
                 node.wait_for_rpc_connection()
 
-            # For backward compatibility of the python scripts with previous
-            # versions of the cache, set mocktime to Jan 1,
-            # 2014 + (201 * 10 * 60)"""
-            self.mocktime = 1388534400 + (201 * 10 * 60)
-
-            # Create a 200-block-long chain; each of the 4 first nodes
+            # Create a 199-block-long chain; each of the 4 first nodes
             # gets 25 mature blocks and 25 immature.
-            # Note: To preserve compatibility with older versions of
-            # initialize_chain, only 4 nodes will generate coins.
-            #
-            # blocks are created with timestamps 10 minutes apart
-            # starting from 2010 minutes in the past
-            block_time = self.mocktime - (201 * 10 * 60)
+            # The 4th node gets only 24 immature blocks.
             for i in range(2):
                 for peer in range(4):
                     for j in range(25):
-                        set_node_times(self.nodes, block_time)
+                        if i == 1 and peer == 3 and j == 24:
```

Comment:
Maybe something more straightforward?
```py
def gen_blocks(peer, blocks):
    node = self.nodes[peer]
    address = node.get_deterministic_priv_key().address
    return node.generatetoaddress(blocks, address)

gen_blocks(0, 25)
gen_blocks(1, 25)
gen_blocks(2, 25)
gen_blocks(3, 24)
```

Reply:
No, that would generate 99 blocks, we need 199.

Not sure if this looks simpler:

```py
gen_blocks(0, 25)
gen_blocks(1, 25)
gen_blocks(2, 25)
gen_blocks(3, 25)
gen_blocks(0, 25)
gen_blocks(1, 25)
gen_blocks(2, 25)
gen_blocks(3, 24)

Reply:
Oh right, I was doing for 99 🤦‍♂️

Reply:
In that case
```py
gen_blocks(0, 50)
gen_blocks(1, 50)
gen_blocks(2, 50)
gen_blocks(3, 49)
```
Or is there a reason against?

Reply:
See the comment two lines up:

```
            # Create a 199-block-long chain; each of the 4 first nodes
            # gets 25 mature blocks and 25 immature.